Source/Core/InputCommon/GCAdapter.cpp

```diff
@@ -453,14 +465,13 @@ void StopScanThread()
 static void Setup()
 {
 #if GCADAPTER_USE_LIBUSB_IMPLEMENTATION
-  const int prev_status = s_status;
+  const AdapterStatus prev_status = s_status;
 
   // Reset the error status in case the adapter gets unplugged
-  if (s_status < 0)
-    s_status = NO_ADAPTER_DETECTED;
+  if (s_status == AdapterStatus::Error)
+    s_status = AdapterStatus::NotDetected;
 
-  s_controller_type.fill(ControllerType::None);
-  s_controller_rumble.fill(0);
+  s_port_states.fill({});
```

Comment:
It doesn't look like you've replaced `s_controller_rumble.fill(0)` with anything. Is this intentional?

Reply:
Good catch. I had the rumble state within `PortState` for a moment but undid that.